static/js/chart/draw.ts

```diff
@@ -427,10 +434,207 @@ function drawLineChart(
+
+  let dataGroups : DataGroup[];
+  dataGroups = Object.values(dataGroupsDict)[0];
+  let legendText = dataGroups.map((dataGroup) => dataGroup.label ? dataGroup.label: '');
+  let colorFn = getColorFn(legendText);
+
+  for (let i = 0; i < dataGroups.length; i++) {
+    styles["colors"].push(colorFn(dataGroups[i].label));
+    styles["statVars"].push(dataGroups[i].label);
+  }
+
+  let dashIndex = 0;
+  let maxV = 0;
+  for (let geoId in dataGroupsDict) {
+    dataGroups = dataGroupsDict[geoId];
+    maxV = Math.max(maxV, Math.max(...dataGroupsDict[geoId].map((dataGroup) => dataGroup.max())));
+    styles["geoIds"].push(geoId);
+    styles["dashes"].push(DASHES[dashIndex]);
+    dashIndex++;
+  }
+  styles["maxV"] = maxV;
+
+  return styles;
+}
+
+
+/**
+ * Draw a group of lines chart with in-chart legend given a dataGroupsDict with different geoIds.
+ *
+ * @param id: DOM id.
+ * @param width: width for the chart.
+ * @param height: height for the chart.
+ * @param dataGroupsDict: {[geoId: string]: DataGroup[]}.
+ * @param unit
+ */
+function drawGroupLineChart(
+    id: string,
+    width: number,
+    height: number,
+    dataGroupsDict: {[geoId: string]: DataGroup[] },
+    unit?: string
+) {
+  // Get all styles.
+  let styles = generateStyles(dataGroupsDict);
+
+  let dataGroups: DataGroup[];
+  dataGroups = Object.values(dataGroupsDict)[0];
+
+  let minV = 0;
+  let maxV = styles["maxV"];
+  // Adjust the width of in-chart legends.
+  let legendWidth = Math.max(width * MARGIN.legendRatio, MARGIN.legendMinWidth);
+
+  let svg = d3
+      .select("#" + id)
+      .append("svg")
+      .attr("width", width)
+      .attr("height", height);
+
+  let xScale = d3
+      .scaleTime()
+      .domain(d3.extent(dataGroups[0].value, (d) => new Date(d.label).getTime()))
+      .range([MARGIN.left, width - MARGIN.right - legendWidth]);
+
+  let yScale = d3
+      .scaleLinear()
+      .domain([minV, maxV])
+      .range([height - MARGIN.bottom, MARGIN.top])
+      .nice(NUM_Y_TICKS);
+
+  addXAxis(svg, height, xScale);
+  addYAxis(svg, width, yScale, unit);
+
+  let dashIndex = 0;
+  for (let geoId in dataGroupsDict) {
+    dataGroups = dataGroupsDict[geoId];
+    for (let i = 0; i < dataGroups.length; i++) {
+      let dataGroup = dataGroups[i];
+      let dataset = dataGroup.value.map(function (dp) {
+        return [new Date(dp.label).getTime(), dp.value];
+      });
+
+      let line = d3
+          .line()
+          .x((d) => xScale(d[0]))
+          .y((d) => yScale(d[1]));
+
+      svg
+          .append("path")
+          .datum(dataset)
+          .attr("class", "line")
+          .style("stroke", styles["colors"][i])
+          .attr("d", line)
+          .attr("stroke-width", "2")
+          .attr("stroke-dasharray", styles["dashes"][dashIndex]);
+    }
+
+    dashIndex++;
+  }
+
+  let legendId = randDomId();
+  svg
+      .append("g")
+      .attr("id", legendId)
+      .attr("transform", `translate(${width - legendWidth}, 50)`);
```

Comment:
make 50 a constant and defined on the top of the file. Normally, all constant number used in the file should be defined as constant values on top

Reply:
Done